internal/services/storage/storage_management_policy_resource.go

```diff
@@ -300,8 +308,11 @@ func resourceStorageManagementPolicyDelete(d *pluginsdk.ResourceData, meta inter
 		return err
 	}
 
-	if _, err = client.Delete(ctx, rid.ResourceGroup, rid.StorageAccountName); err != nil {
-		return err
+	if resp, err := client.Delete(ctx, rid.ResourceGroup, rid.StorageAccountName); err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("waiting for deletion of %s: %+v", rid, err)
```

Comment:
technically this is:

```suggestion
		return fmt.Errorf("deleting %s: %+v", rid, err)
```